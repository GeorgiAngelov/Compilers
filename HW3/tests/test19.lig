type list1 : {head : int, tail : list1 };
type list2 : {head : int, tail : list2 };

var x : list1;
var y : list2 = x;
