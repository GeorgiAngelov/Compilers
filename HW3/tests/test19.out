(decls
   (decl-type:ok (id list1) (type-struct (typed-id (id head) (type-int)) (typed-id (id tail) (type-id list1))))
   (decl-type:ok (id list2) (type-struct (typed-id (id head) (type-int)) (typed-id (id tail) (type-id list2))))
   (decl-var:ok (id x) (type-id list1))
   (decl-var:NONE (id y) (type-id list2) (id:list1 x)))
Function main() not defined or has wrong type!
Typechecking failed!
